core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -136,6 +136,8 @@ private[spark] class TaskSchedulerImpl(
   // IDs of the tasks running on each executor
   private val executorIdToRunningTaskIds = new HashMap[String, HashSet[Long]]
 
+  private val executorsPendingDecommission = new HashMap[String, DecommissionInfo]
```

Comment:
This is not used at all?

Reply:
ok. I see you used it in #29014. But probably it's better to remove it here. You can rebase #29014 later if this PR gets merged.

Reply:
I was debating about this. I think I am now convinced of moving it to #29014 :-).